python/dgl/distributed/dist_graph.py

```diff
@@ -0,0 +1,506 @@
+
+    def __len__(self):
+        return self._shape[0]
+
+    @property
+    def shape(self):
+        ''' Return the shape of the distributed tensor. '''
+        return self._shape
+
+    @property
+    def dtype(self):
+        ''' Return the data type of the distributed tensor. '''
+        return self._dtype
+
+
+class NodeDataView(MutableMapping):
+    """The data view class when G.nodes[...].data is called.
+
+    See Also
+    --------
+    dgl.DGLGraph.nodes
+    """
+    __slots__ = ['_graph', '_graph_name']
+
+    def __init__(self, g, graph_name):
+        self._graph = g
+        self._graph_name = graph_name
+
+    def __getitem__(self, key):
+        return DistTensor(self._graph, _get_ndata_name(key))
+
+    def __setitem__(self, key, val):
+        raise DGLError("DGL doesn't support set items. Please call init_ndata.")
+
+    def __delitem__(self, key):
+        #TODO how to delete data in the kvstore.
+        pass
+
+    def __len__(self):
+        return self._graph.number_of_nodes()
+
+    def __iter__(self):
+        return iter(self._graph._get_all_ndata_names())
+
+    def __repr__(self):
+        name_list = self._graph._get_all_ndata_names()
+        reprs = {}
+        for name in name_list:
+            dtype, shape, _ = self._graph._client.get_data_meta(_get_ndata_name(name))
+            reprs[name] = 'DistTensor(shape={}, dtype={})'.format(str(shape), str(dtype))
+        return repr(reprs)
+
+class EdgeDataView(MutableMapping):
+    """The data view class when G.edges[...].data is called.
+
+    See Also
+    --------
+    dgl.DGLGraph.edges
+    """
+    __slots__ = ['_graph', '_graph_name']
+
+    def __init__(self, graph, graph_name):
+        self._graph = graph
+        self._graph_name = graph_name
+
+    def __getitem__(self, key):
+        return DistTensor(self._graph, _get_edata_name(key))
+
+    def __setitem__(self, key, val):
+        raise DGLError("DGL doesn't support set items. Please call init_ndata.")
+
+    def __delitem__(self, key):
+        #TODO
+        pass
+
+    def __len__(self):
+        return self._graph.number_of_edges()
+
+    def __iter__(self):
+        return iter(self._graph._get_all_edata_names())
+
+    def __repr__(self):
+        name_list = self._graph._get_all_edata_names()
+        reprs = {}
+        for name in name_list:
+            dtype, shape, _ = self._graph._client.get_data_meta(_get_edata_name(name))
+            reprs[name] = 'DistTensor(shape={}, dtype={})'.format(str(shape), str(dtype))
+        return repr(reprs)
+
+
+class DistGraphServer(KVServer):
+    ''' The DistGraph server.
+
+    This DistGraph server extends KVStore server. Its main function is to load the graph data
+    and share the data in its own partition with shared memory so that the DistGraph client
+    can access the data in the local partition with shared memory.
+
+    Parameters
+    ----------
+    server_id : int
+        KVServer's ID (start from 0).
```

Comment:
Try avoid the concept of `KVStore` in the docstring unless KVStore is a user-facing API.